spec/services/waste_exemptions_engine/registration_completion_service_spec.rb

```diff
@@ -10,6 +10,55 @@ module WasteExemptionsEngine
     let(:registration_completion_service) { RegistrationCompletionService.new(new_registration) }
 
     describe "#complete" do
+      it "is idempotent" do
+        expect(WasteExemptionsEngine::Registration.count).to eq(0)
+        RegistrationCompletionService.new(new_registration).complete
+
+        expect { RegistrationCompletionService.new(new_registration).complete }.to raise_error
```

Comment:
RSpec does raise this message when we run this test:

```
WARNING: Using the `raise_error` matcher without providing a specific error or message risks false positives, since `raise_error` will match when Ruby raises a `NoMethodError`, `NameError` or `ArgumentError`, potentially allowing the expectation to pass without even executing the method you are intending to call. Actual error raised was #<RuntimeError: Can't modify frozen hash>. Instead consider providing a specific error class or message.
```

I agree with it, as it looks like it'd be pretty quick to modify the test to include this specific error.

Reply:
Nice one! Will fix that